apps/web-giddh/src/app/shared/advance-receipt-adjustment/services/adjustment-utility.service.ts

```diff
@@ -107,8 +107,86 @@ export class AdjustmentUtilityService {
 
             data.adjustments = data.voucherAdjustments.adjustments;
             delete data.voucherAdjustments;
+        } else {
+            delete data.voucherAdjustments;
+            data.adjustments = [];
         }
 
         return data;
     }
+
+    /**
+     * Returns the request params for invoice-list api
+     *
+     * @param {*} data
+     * @returns {*}
+     * @memberof AdjustmentUtilityService
+     */
+    public getInvoiceListRequest(data: any): any {
+        const debitNoteVoucher = "debit note";
+        const creditNoteVoucher = "credit note";
+        const salesParentGroups = ['revenuefromoperations', 'otherincome'];
+        const purchaseParentGroups = ['operatingcost', 'indirectexpenses'];
+        const debtorCreditorParentGroups = ['sundrydebtors', 'sundrycreditors'];
+        const cashBankParentGroups = ['cash', 'bankaccounts'];
+
+        if (data?.particularAccount?.parentGroups?.length > 0) {
+            if (data?.particularAccount?.parentGroups[0].uniqueName) {
+                data.particularAccount.parentGroups = data?.particularAccount?.parentGroups?.map(group => group.uniqueName);
+            }
+        }
+
+        const isSalesLedger = data?.ledgerAccount?.parentGroups?.some(account => salesParentGroups.includes(account?.uniqueName));
```

Comment:
We can use single for loop here on each of `data?.ledgerAccount?.parentGroups` and `data?.particularAccount?.parentGroups`

```
let isPurchaseLedger, isDebtorCreditorLedger, isCashBankLedger;
data?.ledgerAccount?.parentGroups.forEach(parent => {
    if (purchaseParentGroups.includes(account?.uniqueName)) {
        isPurchaseLedger = true;
    } else if (debtorCreditorParentGroups.includes(account?.uniqueName)) {
        isDebtorCreditorLedger = true;
    } else if (cashBankParentGroups.includes(account?.uniqueName)) {
        isCashBankLedger = true;
    }
})
```